client/consensus/common/src/lib.rs

```diff
@@ -147,6 +147,11 @@ where
 		let hash = params.post_hash();
 		let number = *params.header.number();
 
+		if params.with_state() {
+			// Force imported state finality
```

Comment:
```suggestion
			// Force imported state finality.
			// Required for warp sync. We assume that preconditions have been
			// checked properly and we are importing a finalized block with state.
```
Nit: Maybe a bit more descriptive, feel free to ignore.